format/FormatMultiImage.py

```diff
@@ -228,29 +228,30 @@ def get_imageset(
                 indices=single_file_indices,
             )
 
+            if single_file_indices is None:
+                single_file_indices = list(range(format_instance.get_num_images()))
+
             # If any are None then read from format
-            if [beam, detector, goniometer, scan].count(None) != 0:
+            if not all((beam, detector, goniometer, scan)):
 
                 # Get list of models
-                beam = []
-                detector = []
-                goniometer = []
-                scan = []
-                for i in range(format_instance.get_num_images()):
-                    beam.append(format_instance.get_beam(i))
-                    detector.append(format_instance.get_detector(i))
-                    goniometer.append(format_instance.get_goniometer(i))
-                    scan.append(format_instance.get_scan(i))
-
-            if single_file_indices is None:
-                single_file_indices = list(range(format_instance.get_num_images()))
+                num_images = format_instance.get_num_images()
+                beam = [None for _ in range(num_images)]
+                detector = [None for _ in range(num_images)]
+                goniometer = [None for _ in range(num_images)]
+                scan = [None for _ in range(num_images)]
```

Comment:
Would this not be simpler and clearer?
```suggestion
                beam = [None] * len(num_images)
                detector = [None] * len(num_images)
                goniometer = [None] * len(num_images)
                scan = [None] * len(num_images)
```

Reply:
`num_images` is an integer (or `None`)

Reply:
Yes, `[None] * num_images` is what I meant.

Reply:
Yes, much better, thank you :) (btw see below for performance comparison)

```
$ python -m timeit "a = [None for _ in range(10000)]"
1000 loops, best of 3: 374 usec per loop

$ python -m timeit "a = [None] * 10000"
10000 loops, best of 3: 33 usec per loop
```